### PR TITLE
backends: Enable TestBindings

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes
Enable bind_test.go
* https://github.com/klaytn/klaytn/pull/744/commits/6c21c8be108e1f318bec7637bb915c4d2f3cdaed: remove unnecessary gas limit in bind_test.go
* https://github.com/klaytn/klaytn/pull/744/commits/de1e0fc8a01df424939314d2a1bb999db901fe17: apply https://github.com/klaytn/klaytn/commit/fe8a69f33a8ab71391ed53c534e5a10d407f355d (Enable TestBindings test)
* https://github.com/klaytn/klaytn/pull/744/commits/918ba7de424d765bd2d881ffe8ebdbe948d47cef: fix Interactor and Defaulter tests of TestBindings. This change is due to removement of pending state.

Related PR
* https://github.com/ethereum/go-ethereum/pull/17358
* https://github.com/ethereum/go-ethereum/pull/20311 - https://github.com/ethereum/go-ethereum/pull/20311/commits/f4ec85486a1532594c5009e3c0b9903cbb93e844

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by aidan-kwon** [comment]
It seems to be fine. However, this change still couldn't enable `TestBindings`.
@yoomee1313 Could you add more changes to enable the test. 
You can refer https://github.com/klaytn/klaytn/pull/556/commits/fe8a69f33a8ab71391ed53c534e5a10d407f355d

**Comment by yoomee1313** [comment]
I agree with you @aidan-kwon. Thank you for the link, but I'm still working on it to enable the TestBindings, and it takes some time.
Dear reviewers, I'll convert it to draft and reopen it when it is ready :)

**Comment by yoomee1313** [comment]
@KlaytnDev I reopen the PR, so please review the changes.

File: accounts/abi/bind/bind_test.go [hunk 1]
```diff
@@ -379,26 +394,26 @@ var bindTests = []struct {
...
+			sim := backends.NewSimulatedBackend(blockchain.GenesisAlloc{auth.From: {Balance: big.NewInt(10000000000)}})
+	
 			// Deploy a slice tester contract and execute a n array call on it
 			_, _, slicer, err := DeploySlicer(auth, sim)
 			if err != nil {
 					t.Fatalf("Failed to deploy slicer contract: %v", err)
 			}
 			sim.Commit()
-
+
```
Comment (ehnuje) [inline accounts/abi/bind/bind_test.go:416]: What are these changes? Tabs or whitespaces?

**Review by ehnuje — APPROVED** [review]
LGTM except for minor comment.

**Review by aidan-kwon — APPROVED** [review]

**Review by jeongkyun-oh — DISMISSED** [review]

File: accounts/abi/bind/bind_test.go (hunk 1 above)
Comment (yoomee1313) [reply accounts/abi/bind/bind_test.go]: Those are tabs, and they are useless.. I removed it https://github.com/klaytn/klaytn/pull/744/commits/e86bfb2469fcaf1d710ada5c05f854402bf12d49

**Review by aidan-kwon — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

File: accounts/abi/bind/bind_test.go [hunk 2]
```diff
@@ -962,11 +975,22 @@ func TestBindings(t *testing.T) {
...
+	moder.Dir = pkg
+	if out, err := moder.CombinedOutput(); err != nil {
+		t.Fatalf("failed to convert binding test to modules: %v\n%s", err, out)
+	}
+	pwd, _ := os.Getwd()
+	replacer := exec.Command(gocmd, "mod", "edit", "-replace", "github.com/klaytn/klaytn="+filepath.Join(pwd, "..", "..", "..")) // Repo root
+	replacer.Dir = pkg
+	if out, err := replacer.CombinedOutput(); err != nil {
+		t.Fatalf("failed to replace binding test dependency to current source tree: %v\n%s", err, out)
+	}
```
Comment (winnie-byun) [inline accounts/abi/bind/bind_test.go:989]: Why are we testing `go mod init` and `go mod edit -replace`?

Comment (yoomee1313) [reply accounts/abi/bind/bind_test.go:989]: It's not an actual testing part. `TestBindings` generates mini tests and those tests are defined at `bindTests` variable. Those mini test files are named as `xxx_test.go` at the randomly generated folder. We use go module, so it generates mod file by `go mod init` and `go mod edit -replace` file.

**Review by KimKyungup — APPROVED** [review]
Thanks. :)

**Review by winnie-byun — APPROVED** [review]

**Pull request merged**